### PR TITLE
Drone config

### PR DESCRIPTION
**Pull request by sdarwin**
Drone is a continuous integration framework similar to Travis CI. [The C++ Alliance](https://cppalliance.org/) is offering a hosted Drone server for Boost libraries. Please refer to https://github.com/CPPAlliance/drone-ci for more information and instructions.

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  linux_cxx("TOOLSET=gcc COMPILER=g++ CXXSTD64=03,11,14,17 Job 12", "g++", packages="", buildtype="boost", buildscript="drone", image="cppalliance/droneubuntu2004:1", environment={'TOOLSET': 'gcc', 'COMPILER': 'g++', 'CXXSTD64': '03,11,14,17,03-gnu,11-gnu,14-gnu,17-gnu', 'DRONE_JOB_UUID': '7b52009b64'}, globalenv=globalenv),
+  linux_cxx("TOOLSET=gcc COMPILER=g++ CXXSTD64=03,11,14,17 Job 13", "g++", packages="", buildtype="boost", buildscript="drone", image="cppalliance/droneubuntu2004:1", environment={'TOOLSET': 'gcc', 'COMPILER': 'g++', 'CXXSTD64': '03,11,14,17,03-gnu,11-gnu,14-gnu,17-gnu', 'DRONE_JOB_UUID': 'bd307a3ec3'}, globalenv=globalenv),
+  linux_cxx("TOOLSET=gcc COMPILER=g++ CXXSTD64=03,11,14,17 Job 14", "g++", packages="", buildtype="boost", buildscript="drone", image="cppalliance/droneubuntu2004:1", environment={'TOOLSET': 'gcc', 'COMPILER': 'g++', 'CXXSTD64': '03,11,14,17,03-gnu,11-gnu,14-gnu,17-gnu', 'DRONE_JOB_UUID': 'fa35e19212'}, globalenv=globalenv),
```
Comment (Lastique) [inline .drone.star:31]: These three are supposed to run on non-x86 CPUs, and I don't see this reflected in the arguments.

**Comment by Lastique** [comment]
I don't think I will merge this as it has a dependency on boost-ci.

**Comment by sdarwin** [comment]
> These three are supposed to run on non-x86 CPUs,

It's a good suggestion, many of the boost developers are discussing tests on alternative CPU architectures.   This could be the next enhancement for drone.

> I don't think I will merge this as it has a dependency on boost-ci.

Ok, that is your decision.  

 The link  https://github.com/CPPAlliance/drone-ci includes instructions for removing dependencies, although not recommended because... how will features such as new CPU architectures get added?  Much of the logic (although not all) will go into the main functions.star file.

**Comment by Lastique** [comment]
> how will features such as new CPU architectures get added?

I would like to be in control of what and how is tested, and when support for new compilers is added, and which compiler flags to use, etc.

**Comment by sdarwin** [comment]
> I would like to be in control of what and how is tested, and when support for new compilers is added, and which compiler flags to use, etc. This is why understanding how to support CI configs is important for me.

Most of the settings are in .drone.star, in your local repo, which means you do control them.

**Comment by Lastique** [comment]
I don't think I will be enabling Drone CI - GitHub Actions and AppVeyor cover our testing. Thanks for the submission though.

**Pull request closed without merging**